### PR TITLE
feat(Transition): make `duration` prop more advanced

### PR DESCRIPTION
**Pull request by layershifter**
This PR makes possible to define diffent durations for `hide` and `show` actions.

### CODE REVIEW

File: docs/app/Examples/modules/Transition/Explorers/TransitionExampleTransitionExplorer.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const transitions = ['jiggle', 'flash', 'shake', 'pulse', 'tada', 'bounce']
 
 const options = transitions.map(name => ({ key: name, text: name, value: name }))
 
-export default class TransitionExampleStaticExplorer extends Component {
+export default class TransitionExampleTransitionExplorer extends Component {
```
Comment (layershifter) [inline docs/app/Examples/modules/Transition/Explorers/TransitionExampleTransitionExplorer.js:8]: Name example's class as file

File: docs/app/Examples/modules/Transition/Explorers/index.js [hunk 2]
```diff
@@ -1,11 +1,10 @@
 import React from 'react'
+import { Message } from 'semantic-ui-react'
```
Comment (layershifter) [inline docs/app/Examples/modules/Transition/Explorers/index.js:2]: Sort imports.

File: docs/app/Examples/modules/Transition/Explorers/index.js [hunk 3]
```diff
@@ -1,11 +1,10 @@
...
 
 import ComponentExample from 'docs/app/Components/ComponentDoc/ComponentExample'
 import ExampleSection from 'docs/app/Components/ComponentDoc/ExampleSection'
 
-import { Message } from 'semantic-ui-react'
-
-const TransitionTypesExamples = () => (
+const TransitionExplorersExamples = () => (
```
Comment (layershifter) [inline docs/app/Examples/modules/Transition/Explorers/index.js:7]: Typo.

File: src/modules/Transition/Transition.js [hunk 4]
```diff
@@ -25,7 +31,13 @@ export default class Transition extends Component {
     children: PropTypes.element.isRequired,
 
     /** Duration of the CSS transition animation in milliseconds. */
-    duration: PropTypes.number,
+    duration: PropTypes.oneOfType([
+      PropTypes.number,
```
Comment (levithomason) [inline src/modules/Transition/Transition.js:35]: Looks like we technically also support a string here.

File: src/modules/Transition/Transition.d.ts [hunk 5]
```diff
@@ -15,7 +15,7 @@ export interface TransitionProps {
   children?: React.ReactNode;
 
   /** Duration of the CSS transition animation in milliseconds. */
-  duration?: number;
+  duration?: number | TransitionPropDuration;
```
Comment (levithomason) [inline src/modules/Transition/Transition.d.ts]: String is also supported, correct?

File: src/modules/Transition/TransitionGroup.d.ts [hunk 6]
```diff
@@ -14,7 +16,7 @@ export interface TransitionGroupProps {
   children?: React.ReactNode;
 
   /** Duration of the CSS transition animation in milliseconds. */
-  duration?: number;
+  duration?: number | TransitionPropDuration;
```
Comment (levithomason) [inline src/modules/Transition/TransitionGroup.d.ts]: Also string I believe.

File: src/modules/Transition/TransitionGroup.js [hunk 7]
```diff
@@ -31,7 +31,13 @@ export default class TransitionGroup extends React.Component {
     children: PropTypes.node,
 
     /** Duration of the CSS transition animation in milliseconds. */
-    duration: PropTypes.number,
+    duration: PropTypes.oneOfType([
+      PropTypes.number,
```
Comment (levithomason) [inline src/modules/Transition/TransitionGroup.js:35]: Finally, let's add string as a supported type here too.

**Comment by levithomason** [comment]
I've rebased to the latest master and added string duration propType / typings support.  Will merge on pass.

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.71.5`

**Pull request merged**